plugins/shell-proxy/ssh-proxy.py

```diff
@@ -22,7 +22,7 @@
 
 def make_argv():
     yield "nc"
-    if sys.platform == 'linux':
+    if sys.platform in {'linux', 'cygwin'}:
         # caveats: macOS built-in netcat command not supported proxy-type
```

Comment:
plz update the caveats

Reply:
I will update the comments tomorrow (because I found that there exists several netcat versions on Linux, such as `netcat-openbsd` (default, works), `netcat-traditional` and `ncat` (nmap netcat)).

Reply:
I've examined the preinstalled versions of netcat on some Linux distributions, MacOS, and Cygwin. It seems that all of them now support the '-X' option. For a more general check, I think the following modification is better. I have tested it.

```diff
diff --git a/plugins/shell-proxy/ssh-proxy.py b/plugins/shell-proxy/ssh-proxy.py
index 5d239177..e68d109e 100755
--- a/plugins/shell-proxy/ssh-proxy.py
+++ b/plugins/shell-proxy/ssh-proxy.py
@@ -22,7 +22,7 @@ if parsed.scheme not in proxy_protocols:
 
 def make_argv():
     yield "nc"
-    if sys.platform in {'linux', 'cygwin'}:
+    if b'-X ' in subprocess.check_output(['nc', '-h'], stderr=subprocess.STDOUT):
         # caveats: macOS built-in netcat command not supported proxy-type
         yield "-X" # --proxy-type
         # Supported protocols are 4 (SOCKS v4), 5 (SOCKS v5) and connect (HTTP proxy).
```

Reply:
I've added a caveat.

Reply:
I also investigated the netcat versions on linux, cygwin and macos, and found that they all support the '-X proto' option. I tried to make it more general by using `if b'-X ' in subprocess.run(['nc', '-h'], check=False, stderr=subprocess.STDOUT, stdout=subprocess.PIPE).stdout:`. However, I failed to clone my repo on MacOS.